machine/fdt.c

```diff
@@ -144,6 +144,22 @@ int fdt_string_list_index(const struct fdt_scan_prop *prop, const char *str)
   return -1;
 }
 
+void fdt_version_prop_print(const uint32_t * version_value, int version_len)
+{
+  const char *char_data = (char *)version_value;
+  if (!char_data) {
+    char_data = "unknown";
+    version_len = strlen(char_data) + 1;
+  }
+  // The size should be 1, but print any extra values if they appear
```

Comment:
This doesn't quite make sense; the size would normally be strlen+1, which is not 1. Maybe "There should only be one string, but print any extra strings if they appear"?